tests/unittests/test_child_bookmarks.py

```diff
@@ -67,20 +77,92 @@ def test_sync_event_updates_when_events_bookmark_present(self, mock_sync_event_u
+    @mock.patch('tap_stripe.convert_dict_to_stripe_object')
+    @mock.patch('tap_stripe.is_parent_selected', return_value=False)
+    @mock.patch('tap_stripe.paginate', return_value = [mock.Mock()])
+    @mock.patch('tap_stripe.Context.is_sub_stream', return_value=False)
+    @mock.patch('tap_stripe.singer.write_schema')
+    @mock.patch('tap_stripe.Context.is_selected', return_value=[True])
+    @mock.patch('tap_stripe.metadata.to_map')
+    @mock.patch('tap_stripe.Context.get_catalog_entry')
+    @mock.patch('tap_stripe.utils.now')
+    @mock.patch('tap_stripe.sync_sub_stream')
+    @mock.patch('tap_stripe.sync_event_updates')
+    @mock.patch('tap_stripe.get_bookmark_for_sub_stream')
+    @mock.patch('tap_stripe.singer.write_record')
+    def test_sync_event_updates_for_parent_stream(self, mock_write_record, mock_get_bookmark, 
+                                                                mock_sync_event_updates, mock_sync_sub_stream, 
+                                                                mock_now, mock_get_catalog_entry, 
+                                                                mock_to_map, mock_is_selected, mock_write_schema, 
+                                                                mock_is_sub_stream, mock_paginate, mock_is_parent_selected, 
+                                                                mock_convert_dict, mock_reduce_foreign_keys):
+        '''
+            Verify that when only the parent stream is selected, write_record() is called for the parent stream.
+        '''
+        # mocked now time
+        now_time = utils.strptime_with_tz('2022-01-31 16:17:40.948019+00:00')
+        mock_get_bookmark.return_value = 1641137550
+        mock_now.return_value = now_time
+        # catalog passed in the context
+        Context.catalog = {'streams': [{'tap_stream_id': 'invoices', 'schema': {}, 'key_properties': [], 'metadata': []}]}
+        Context.config = {'start_date': "2019-06-20 16:17:40.948019+00:00"}
+        mock_get_catalog_entry.return_value = {'tap_stream_id': 'invoices', 'schema': {}, 'key_properties': [], 'metadata': [{"valid-replication-keys": ["created"]}]}
+        # metadata.to_map return value
+        mock_to_map.return_value = {(): {'table-key-properties': ['id'], 'selected': True, 'forced-replication-method': 'INCREMENTAL', 'valid-replication-keys': ['created']}}
+        invoice_line_items_ts = 1641137533 # 02-01-2022T03:32:13Z
+        events_ts = 1645716195
+        Context.state = {"bookmarks": {"invoice_line_items_events": {"updates_created": events_ts}, "invoice_line_items": {"date": invoice_line_items_ts}}}
+        sync()
+
+        # Verify that one record is being written
+        self.assertEqual(1, mock_write_record.call_count)
+
+    @mock.patch('tap_stripe.metadata.to_map')
+    @mock.patch('tap_stripe.Context.get_catalog_entry')
+    def test_is_parent_selected_when_parent_is_selected(self, mock_get_catalog_entry, mock_to_map):
```

Comment:
Do you mean when sub strem selected? If yes then give test name accordingly

Reply:
Updated the testcase name